### DISCUSSION TITLE
Getting result type of expression at compile time

### DISCUSSION THREAD

**Discussion by purplefox** [antonmedv/expr#363] [General]
Hi,

Is there any way to get the return type of an expression at the compile stage (i.e. without evaluating it)?

**Comment by antonmedv** [comment]
You can get the return type by checker.Check.

**Reply by purplefox** [reply]
I am using `Compile` to compile an expression string to a `*vm.Program`.

Given a `*vm.Program` how do I get a reference to `*parser.Tree` so I can call `Checker.Check`?

I'm guessing I will need to implement my own Compile method, which parses, checks then calls Compile with the parse tree?

TIA

**Reply by antonmedv** [reply]
```go
		env := map[string]interface{}{
			"Any": interface{}("foo"),
		}

		tree, err := parser.Parse("Any")
		require.NoError(t, err)

		config := conf.New(env)
		expr.AsKind(reflect.String)(config)

		_, err = checker.Check(tree, config)
		require.NoError(t, err)
```

Probably we can extend Compile method to CompileWithTime
